src/Client/GuzzleFileFetcher.php

```diff
@@ -84,8 +84,10 @@ public function fetchMetaData(string $fileName, int $maxBytes): PromiseInterface
     /**
      * {@inheritDoc}
      */
-    public function fetchTarget(string $fileName, int $maxBytes, array $options = []): PromiseInterface
+    public function fetchTarget(string $fileName, int $maxBytes, array $options = [], string $url = null): PromiseInterface
     {
+        // Allow calling code to download the target from an arbitrary URL.
+        $fileName = $url ?? $fileName;
```

Comment:
Should we assign to a new more generic var like `$location` or something?

Reply:
Nice idea. Renamed to `$location`.